### PR TITLE
[Mavericks] Update PHPUnit for Functional tests

### PR DESCRIPTION
**Pull request by rsavchuk**
### Story
* [MTA-3311](https://jira.corp.magento.com/browse/MTA-3311) Upgrade PHPUnit to the Latest Version

### Task
* [MTA-3484](https://jira.corp.magento.com/browse/MTA-3484) Resolve an issue with upgrade to the latest Selenium and FF

### Bamboo CI builds
* [ ] [Functional Acceptance Tests](https://bamboo.corp.magento.com/browse/M2-FAT537/latest)
* [ ] [Functional Unstable Tests](https://bamboo.corp.magento.com/browse/M2-FUT167/latest)
* [ ] [Sample Data Tests](https://bamboo.corp.magento.com/browse/M2-SDT440/latest)

### Related Pull Requests
- https://github.com/magento-qmt/magento2ce/pull/7
- https://github.com/magento-qmt/magento2ee/pull/5
- https://github.com/magento-qmt/magento2-infrastructure/pull/2

### Checklist
- [ ] Jira issues have accurate summary, meaningful description and have links to relevant documentation at the story/task level (if applicable)
- [ ] Stories approved by product owner

### CODE REVIEW

**Comment by rsavchuk** [comment]
### Story
* [MTA-3311](https://jira.corp.magento.com/browse/MTA-3311) Upgrade PHPUnit to the Latest Version

### Task
* [MTA-3484](https://jira.corp.magento.com/browse/MTA-3484) Resolve an issue with upgrade to the latest Selenium and FF

### Bamboo CI builds

* [ ] [M2 L3 (integrity, static legacy, static)](https://bamboo.corp.magento.com/browse/M2-L3415/latest)
* [ ] [Functional Acceptance Tests](https://bamboo.corp.magento.com/browse/M2-FAT422/latest)
* [ ] [Functional Unstable Tests](https://bamboo.corp.magento.com/browse/M2-FUT71/latest)
* [ ] [Sample Data Tests](https://bamboo.corp.magento.com/browse/M2-SDT344/latest)
* [ ] [Semantic Version Checker](https://bamboo.corp.magento.com/browse/SVC-SVC240/latest)



### Related Pull Requests
https://github.com/magento-qmt/magento2ce/pull/7
https://github.com/magento-qmt/magento2ee/pull/5
https://github.com/magento-qmt/magento2-infrastructure/pull/2

### Checklist
- [ ] PR is green on M2 Quality Portal
- [ ] Jira issues have accurate summary, meaningful description and have links to relevant documentation at the story/task level (if applicable)
- [ ] Stories approved by product owner
- [ ] Backward incompatible and/or other important for community changes are documented (if applicable)
- [ ] Semantic Version build failure is approved by architect (if build is red). Architect Name:____________________________________________

File: Magento/Mtf/TestCase/Functional.php [hunk 1]
```diff
@@ -633,52 +701,6 @@ public function setOutputCallback($callback)
...
-     * @param  string $mockClassName Class name for the generated test double class.
-     * @param  boolean $callOriginalConstructor Can be used to disable the call to the original class' constructor.
-     * @param  boolean $callOriginalClone Can be used to disable the call to the original class' clone constructor.
-     * @param  boolean $callAutoload Can be used to disable __autoload() during the generation of the test double class.
-     * @param  boolean $cloneArguments
-     * @param  boolean $callOriginalMethods
-     * @return \PHPUnit_Framework_MockObject_MockObject
-     * @throws \PHPUnit_Framework_Exception
-     */
-    public function getMock(
```
Comment (slavvka) [inline Magento/Mtf/TestCase/Functional.php]: Isn't it a backward incompatible change? Was it documented somewhere?

File: Magento/Mtf/TestRunner/Process/TestResult.php [hunk 2]
```diff
@@ -29,7 +29,7 @@ class TestResult extends \PHPUnit_Framework_TestResult
      * @param float $time
      * @return void
      */
-    public function addError(\PHPUnit_Framework_Test $test, \Exception $e, $time)
+    public function addError(\PHPUnit_Framework_Test $test, $e, $time)
```
Comment (slavvka) [inline Magento/Mtf/TestRunner/Process/TestResult.php]: Isn't it a backward incompatible change? Was it documented somewhere?

File: Magento/Mtf/TestRunner/Process/TestResult.php [hunk 3]
```diff
@@ -135,10 +139,12 @@ public function addFailure(\PHPUnit_Framework_Test $test, \PHPUnit_Framework_Ass
...
 
     /**
-     * @param \Exception $exception
+     * Wrap an exception.
+     * 
+     * @param \Throwable $exception
      * @return Failure|Incomplete|Risky|Skipped
      */
-    protected function wrapException(\Exception $exception)
+    protected function wrapException(\Throwable $exception)
```
Comment (slavvka) [inline Magento/Mtf/TestRunner/Process/TestResult.php]: Isn't it a backward incompatible change? Was it documented somewhere?

**Comment by lanken** [comment]
Have you ensured that provided changes are compatible both with PHPUnit 4.8 and PHPUnit 5.5?

**Comment by Rykh** [comment]
Checked on PHPUnit 4.8.27 and PHP 7.0.8 new faile tests not founded

**Comment by lanken** [comment]
Approve changes related to the upgrading PHPUnit version usage in the MTF.

**Comment by slavvka** [comment]
Approved

**Pull request merged**